sources/core/objects/turfs/MetalWall.h

```diff
@@ -1,26 +1,47 @@
 #pragma once
 
 #include "Turf.h"
+#include <bitset>
+#include <array>
 
 namespace kv
 {
 
+const std::array<Dir, 4> WALL_PROCESSING_DIRS = {Dir::NORTH, Dir::SOUTH, Dir::WEST, Dir::EAST};
+
 class MetalWall : public Turf
 {
 public:
     DECLARE_SAVEABLE(MetalWall, Turf);
     REGISTER_CLASS_AS(MetalWall);
+
     MetalWall();
+
+    void UpdateState(Dir dir = Dir::ALL);
+    void UpdateNeighborhoodState(Dir dir = Dir::ALL);
+    void CheckNeighborhood(Dir dir = Dir::ALL);
```

Comment:
These functions are called only from `MetalWall`, so they should be made private.